lib/puppet/pops/evaluator/deferred_resolver.rb

```diff
@@ -16,10 +16,12 @@ class DeferredResolver
   #
   # @param facts [Puppet::Node::Facts] the facts object for the node
   # @param catalog [Puppet::Resource::Catalog] the catalog where all deferred values should be replaced
+  # @param environment [Puppet::Node::Environment] the environment whose anonymous module methods
+  #  are to be mixed into the scope
   # @return [nil] does not return anything - the catalog is modified as a side effect
   #
-  def self.resolve_and_replace(facts, catalog)
-    compiler = Puppet::Parser::ScriptCompiler.new(catalog.environment_instance, catalog.name, true)
+  def self.resolve_and_replace(facts, catalog, environment)
```

Comment:
Apparently this method is called by things other than puppet. I'll make the environment parameter optional and default to `catalog.environment_instance` like it did before.

```
13:45:09     Received response from pcp://wily-marking.delivery.puppetlabs.net/agent: {:envelope=>{:id=>"be6b2cf2-d339-4f31-8f22-4caa698c0cb2", :message_type=>"http://puppetlabs.com/rpc_blocking_response", :sender=>"pcp://wily-marking.delivery.puppetlabs.net/agent", :targets=>["pcp://jnlp-r95x2/ruby-pcp-client-1094"], :expires=>"1970-01-01T00:00:00.000Z"}, :data=>{"transaction_id"=>"73ae19db-dea5-4dcf-b30d-caf2a3f56616", "results"=>{"transaction_id"=>"925d4b26-b7e7-41ad-8571-c672b119e65f", "exitcode"=>1, "status"=>"failure", "stderr"=>"/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/pops/evaluator/deferred_resolver.rb:23:in `resolve_and_replace': wrong number of arguments (given 2, expected 3) (ArgumentError)\n\tfrom /opt/puppetlabs/pxp-agent/tasks-cache/apply_ruby_shim/apply_ruby_shim.rb:125:in `block in <main>'\n\tfrom /opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/context.rb:62:in `override'\n\tfrom /opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet.rb:314:in `override'\n\tfrom /opt/puppetlabs/pxp-agent/tasks-cache/apply_ruby_shim/apply_ruby_shim.rb:121:in `<main>'\n"}}}
```